databricks/koalas/base.py

```diff
@@ -299,10 +300,23 @@ def is_monotonic(self):
 
         >>> ser.rename("a").to_frame().set_index("a").index.is_monotonic
         True
+
+        >>> ser = ks.Series([5, 4, 3, 2, 1], index=[1, 2, 3, 4, 5])
+        >>> ser.is_monotonic
+        False
+
+        >>> ser.index.is_monotonic
+        True
         """
+        from databricks.koalas.frame import DataFrame
+        from databricks.koalas.series import _col
+        if self.hasnans:
+            return False
         col = self._scol
-        window = Window.orderBy(self._kdf._internal.index_scols).rowsBetween(-1, -1)
-        return self._with_new_scol((col >= F.lag(col, 1).over(window)) & col.isNotNull()).all()
+        sdf = self._internal.sdf.withColumn('id', monotonically_increasing_id())
```

Comment:
Shall we use more complex name like `__id__` or something? `id` will conflict with user columns.

Reply:
thanks for commenting! resolved! 😄 
(fix to use `__index_level_temp__`)